internal/xds/translator/translator.go

```diff
@@ -461,5 +458,53 @@ const (
 const (
 	DefaultEndpointType EndpointType = iota
 	Static
+	DNS
 	EDS
 )
+
+// splitEndpointsByHostAddressType splits FQDN host address type DestinationEndpoints from DestinationSettings.
+func splitEndpointsByHostAddressType(rd *ir.RouteDestination, lb *ir.LoadBalancer, protocol ProtocolType) []*xdsClusterArgs {
+	// Group DestinationSettings by the type of endpoint.
+	groups := make(map[EndpointType][]*ir.DestinationSetting)
+	groupOrders := []EndpointType{Static, DNS}
+
+	for _, ds := range rd.Settings {
+		var fqdn, ips []*ir.DestinationEndpoint
+		for _, ep := range ds.Endpoints {
+			if ep.Type == discoveryv1.AddressTypeFQDN {
+				fqdn = append(fqdn, ep)
+			} else {
+				ips = append(ips, ep)
+			}
+		}
+		if len(fqdn) > 0 {
+			groups[DNS] = append(groups[DNS], &ir.DestinationSetting{
+				Weight:    ds.Weight,
+				Endpoints: fqdn,
+			})
+		}
+		if len(ips) > 0 {
+			groups[Static] = append(groups[Static], &ir.DestinationSetting{
+				Weight:    ds.Weight,
```

Comment:
From this implementation, you have already used weight to balance FQDN and IP. 
@arkodg Do we still want to go to this [simple solution](https://github.com/envoyproxy/gateway/issues/1922#issuecomment-1806292224)?

Reply:
hi @EltonzHu i am sticking to the simple approach in this PR, PTAL